Composer/packages/client/src/recoilModel/selectors/project.ts

```diff
@@ -71,9 +81,47 @@ export const botProjectSpaceSelector = selector({
       const formDialogSchemas = get(formDialogSchemasSelectorFamily(projectId));
       const metaData = get(projectMetaDataState(projectId));
       const botError = get(botErrorState(projectId));
+      const buildEssentials = get(buildEssentialsSelector(projectId));
       const name = get(botDisplayNameState(projectId));
       const botNameId = get(botNameIdentifierState(projectId));
-      return { dialogs, formDialogSchemas, projectId, name, ...metaData, error: botError, botNameId };
+
+      const luFiles = get(luFilesState(projectId));
```

Comment:
This selector is consumed by a lot of components. The issue with this core selector being big is any changes to lu, settings qna files etc is going to rerender all the components leveraging this selector (for ex filepersistence). My view is make these core selectors fetch as little as possible and the really big selectors are just used by one off components.

For ex botProjectDiagnosticsSelector makes sense that it gets all those values but can we avoid localBotsWithoutErrorsSelector from being this big?

Reply:
I searched there are two usage of `botProjectSpaceSelector`, 
1. ProjectTree, cause we want errors/warnings status display on treeItem.
2. Notification Page, also gathering all skills error/warning.
It seems they both need this diagnostics, and no calculation waste here ?